torch/csrc/autograd/engine.cpp

```diff
@@ -135,6 +136,8 @@ struct GraphTask {
   bool keep_graph_;
   bool grad_mode_;
 
+ //  Protects access to captured_vars_, and for use by not_done_
+ //  The only writes to exec_info are in init_to_execute()
```

Comment:
Maybe it'll be easier to find this information if we just state in `exec_info_` that this is immutable (and thus thread-safe to read without synchronization.) Are the other fields mutable, in that case?

Reply:
But actually, it looks like `init_to_execute` is called why we have the `graph_task.mutex_` (see line 39) so it's probably easier to just say that it is also protected by the mutex.

Reply:
It feels unlikely that the other non-atomic fields in this struct aren't also protected by `mutex_`. Worth double checking?

Reply:
(One way to check these things: look for accesses of the members, and then check and see if the lock was taken out at some point above.)